app/v5/controllers/validators/resolvers/ResolveOnlyOneJsonProperty.scala

```diff
@@ -0,0 +1,35 @@
+/*
+ * Copyright 2023 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package v5.controllers.validators.resolvers
+
+import cats.data.Validated
+import cats.data.Validated.{Invalid, Valid}
+import play.api.libs.json.JsValue
+import shared.models.errors.MtdError
+import v5.models.errors.RuleBothPropertiesSuppliedError
+
+class ResolveOnlyOneJsonProperty(fieldOneName: String, fieldTwoName: String) {
```

Comment:
This one seems fairly gen-purpose, could go in the shared resolvers?

Reply:
Moved to shared and renamed `ResolveExclusiveJsonProperty`. Also:

-  simplified its usage so no need to use `productR`. This reverts behaviour for edge case where both properties are supplied _and_ they also happen to be empty to what it used to be (see [here](https://github.com/hmrc/self-assessment-bsas-api/blob/0433b3fc587749c281f62db7dfe824b7c457a9d9/test/v3/controllers/requestParsers/validators/SubmitUkPropertyBsasValidatorSpec.scala#L231)).
- Moved resolver instance creation to the validator companions so that they can be shared between requests.